tests/testthat/test_lightgbm_1.R

```diff
@@ -0,0 +1,237 @@
+context("test tidiers for LightGBM (training and test data)")
+
+testthat::skip_if_not_installed("lightgbm")
+
+testdata_dir <- tempdir()
+testdata_filename <- "airline_2013_10_tricky_v3_5k.csv"
+testdata_file_path <- paste0(testdata_dir, "/", testdata_filename)
+
+filepath <- if (!testdata_filename %in% list.files(testdata_dir)) {
+  "https://exploratory-download.s3.us-west-2.amazonaws.com/test/airline_2013_10_tricky_v3.csv"
+} else {
+  testdata_file_path
+}
+
+flight <- exploratory::read_delim_file(filepath, ",", quote = "\"", skip = 0, col_names = TRUE, na = c("", "NA"), locale = readr::locale(encoding = "UTF-8", decimal_mark = "."), trim_ws = FALSE, progress = FALSE) %>%
+  exploratory::clean_data_frame()
+
+if (!testdata_filename %in% list.files(testdata_dir)) {
+  set.seed(1)
+  flight <- flight %>% slice_sample(n = 5000)
+  write.csv(flight, testdata_file_path)
+}
+
+test_that("exp_lightgbm(regression) evaluate training and test with FIRM importance", {
+  set.seed(1)
+  model_df <- flight %>%
+    exp_lightgbm(`ARR DELAY`, `CAR RIER`, `ORI GIN`, `DEP DELAY`, `AIR TIME`, `FL DATE`,
+      predictor_funs = list(`CAR RIER` = "none", `ORI GIN` = "none", `DEP DELAY` = "none", `AIR TIME` = "none", list(`FL DATE_y` = "year", `FL DATE_m` = "monname", `FL DATE_dom` = "day", `FL DATE_dow` = "wday")),
+      test_rate = 0.3,
+      test_split_type = "ordered", pd_with_bin_means = TRUE,
+      watchlist_rate = 0.1,
+      importance_measure = "firm"
+    )
+
+  ret <- model_df %>% prediction(data = "training_and_test", pretty.name = TRUE)
+
+  ret <- flight %>% select(-`ARR DELAY`) %>% add_prediction(model_df = model_df)
+  ret <- model_df %>% prediction(data = "newdata", data_frame = flight)
+
+  ret <- model_df %>% tidy_rowwise(model, type = "evaluation_log")
+  ret <- model_df %>% prediction(data = "training_and_test")
+  test_ret <- ret %>% filter(is_test_data == TRUE)
```

Comment:
Add a brief comment for each line that does the result check. What is the number representing and why that range if it uses lt or gt.